Pull request for issue #9, issue #42 and css/template improvements

byteme opened the pull request:
1) Added registration, login and logout page (issue #9).
2) Added redirect to previous page after login (issue #42).
3) Added redirect to / if logged in user visits register or login page
4) Added site navigation.
5) Improved style.css and added ie-fix.css.
6) Few changes in structure (base) templates.

Updated pull request:

1) Changed function views into class-based views
2) Navigation now uses functions instead of hard coded string to generate links
3) Changed strings into translation strings
4) Forms are now generated by form.as_p
5) Changed license & copyright text in footer
6) Removed uneccessary imports
7) Removed Logout.html
8) Changed fulltitle block into title in register.html and login.html
9) Switched hardcoded url redirects with functions
10) Changed some spacing in coding style

Updated pull request:
1) Fixed remaining strings with trans strings
2) Now importing only modules (w/o content - hopefully :))

Updated pull request:

1) Fixed css alignment
2) Alphabetically ordered imports
3) Changed forms import part to absolute path
4) Removed copyright div from footer

byteme committed:
added registration, login, logout and few html/css changes

byteme committed:
1) added redirect to previous page after login (issue #42)
2) added redirect to main page in case logged in user tries to access register/login page
3) removed same uneccessary variables
4) reworked default css style and added ie-fix.css
5) added navigation template (nav.html)
6) minor changes in basic templates

mitar commented:
Which team are you? Please move yourself to Team 4.

mitar commented on piplmesh/urls.py:
```diff
@@ -10,4 +10,10 @@
     url(r'^admin/doc/', include('django.contrib.admindocs.urls')),
     url(r'^admin/', include(admin.site.urls)),
+
+    # Auth
```
Please spell comments out. _Auth_ -> _Authentication_

mitar commented on piplmesh/frontend/templates/footer.html:
```diff
@@ -1,4 +1,3 @@
-        <p>Footer here!</p>
-    </div>
+<div id="footer">
+    <p>PiplMesh &copy; 2012</p>
```
Please write similar copyright notice as it is on [wlan slovenija](http://wlan-si.net/) home page (you can link to the same page there).

mitar commented on piplmesh/frontend/static/piplmesh/css/style.css:
```diff
@@ -1,64 +1,96 @@
+}
+
+.error {
+    color:#ff0000;
```
Write space after `:`.

mitar commented on piplmesh/account/views.py:
```diff
@@ -0,0 +1,50 @@
+        return HttpResponseRedirect("/")
+
+    form = AuthenticationForm()
+    data = {'form':form}
```
Space after `:`.

mitar commented on piplmesh/account/views.py:
```diff
@@ -0,0 +1,50 @@
+
+def login_view(request):
+    if request.user.is_authenticated():
+        return HttpResponseRedirect("/")
```
Why redirect to `/`? Why not back to the site from where user came from? Or at least to the `home` URL (as specified in `urls.py`). Do not hard-code things!

mitar commented on piplmesh/account/views.py:
```diff
@@ -0,0 +1,50 @@
+
+def logout_view(request):
+    logout(request)
+    return HttpResponseRedirect("/")
```
Same as above.

mitar commented on piplmesh/account/views.py:
```diff
@@ -0,0 +1,50 @@
+from django.contrib.auth import authenticate, login
+from forms import RegisterForm
+
+def register_view(request):
```
You should really learn to use class-based views.

mitar commented on piplmesh/account/views.py:
```diff
@@ -0,0 +1,50 @@
+    else:
+        form = RegisterForm()
+
+    data = {'form':form}
```
Space after `:`.

mitar commented on piplmesh/account/views.py:
```diff
@@ -0,0 +1,50 @@
+from django.contrib.auth.forms import UserCreationForm
```
Import only modules, not directly their content.

mitar commented on piplmesh/account/templates/register.html:
```diff
@@ -0,0 +1,43 @@
+        </div>
+
+        <div>
+            <label for="id_password2">Repeat password:</label>
```
All strings should be [translatable](https://docs.djangoproject.com/en/dev/topics/i18n/).

mitar commented on piplmesh/account/templates/register.html:
```diff
@@ -0,0 +1,43 @@
+{% extends "plain.html" %}
+
+{% block fulltitle %}Register{% endblock %}
```
Why not `title` block?

mitar commented on piplmesh/account/templates/logout.html:
```diff
@@ -0,0 +1,9 @@
+
+{% block content %}
+<div class="main">
+    Logout
```
Please be more verbose.

mitar replied:
User should understand what has happened.

mitar commented on piplmesh/account/templates/login.html:
```diff
@@ -0,0 +1,23 @@
+    <h1>Login</h1>
+
+    <form action="" method="post">
+        {% if error %}<div class="error">Your username and password didn't match. Please try again.</div>{% endif %}
```
Translatable.

mitar replied:
I am not really fan of manually done forms. This should be done automatically from the Django form object. Not that you have to keep in sync fields in template with fields in the form all the time.

mitar commented on piplmesh/account/templates/login.html:
```diff
@@ -0,0 +1,23 @@
+        {% if error %}<div class="error">Your username and password didn't match. Please try again.</div>{% endif %}
+        {% csrf_token %}
+        <div>
+            <label for="id_username">Username:</label>
```
Translatable.

mitar commented on piplmesh/account/forms.py:
```diff
@@ -0,0 +1,20 @@
+from django.contrib.auth.forms import UserCreationForm
```
Import only modules.

mitar replied:
TODO.

mitar replied:
Still TODO.

byteme committed:
1) Changed function views into class-based views
2) Navigation now uses functions instead of hard coded string to generate links
3) Changed strings into translation strings
4) Forms are now generated by form.as_p
5) Changed license & copyright text in footer
6) Removed uneccessary imports
7) Removed Logout.html
8) Changed fulltitle block into title in register.html and login.html
9) Switched hardcoded url redirects with functions
10) Changed some spacing in coding style

byteme committed:
Removed auto login after registration (email activation should be added)

byteme commented:
I've updated pull request some time ago (forgot to make a comment here).

mitar commented on piplmesh/account/forms.py:
```diff
@@ -0,0 +1,20 @@
+
+    class Meta:
+        model = User
+        fields = ("username", "email", )
```
Clean comma at the end. Write commas after last element only if you split tuple into multiple lines.

mitar commented on piplmesh/account/forms.py:
```diff
@@ -0,0 +1,20 @@
+        fields = ("username", "email", )
+
+    def clean_email(self):
+        email = self.cleaned_data["email"]
```
For constant strings use `'`.

mitar commented on piplmesh/account/templates/login.html:
```diff
@@ -0,0 +1,16 @@
+{% extends "plain.html" %}
+{% load i18n %}
```
One empty line before `load`, to have a separate section for loading.

mitar commented on piplmesh/account/templates/login.html:
```diff
@@ -0,0 +1,16 @@
+{% extends "plain.html" %}
+{% load i18n %}
+
+{% block title %}Login{% endblock %}
```
Translatable string.

mitar commented on piplmesh/account/templates/login.html:
```diff
@@ -0,0 +1,16 @@
+    <form action="" method="post">
+        {% csrf_token %}
+        {{ form.as_p }}
+        <input type="submit" value="Login" />
```
Translatable string.

mitar commented on piplmesh/account/templates/register.html:
```diff
@@ -0,0 +1,16 @@
+{% extends "plain.html" %}
+{% load i18n %}
+
+{% block title %}Register{% endblock %}
```
Translatable string.

mitar commented on piplmesh/account/templates/register.html:
```diff
@@ -0,0 +1,16 @@
+{% extends "plain.html" %}
+{% load i18n %}
```
Empty line before.

mitar commented on piplmesh/account/templates/register.html:
```diff
@@ -0,0 +1,16 @@
+    <form action="" method="post">
+        {% csrf_token %}
+        {{ form.as_p }}
+        <input type="submit" value="Register" />
```
Translatable string.

mitar commented on piplmesh/account/views.py:
```diff
@@ -0,0 +1,46 @@
+from django.core.urlresolvers import reverse
```
Import only modules.

byteme replied:
How to import only module and not content? I've googled it but I couldn't find anything. :/

mitar replied:
```
from django.core import urlresolvers
```

and then use:

```
urlresolvers.reverse(...)
```

byteme replied:
Thanks, gonna fix them shorty.

mitar commented on piplmesh/account/views.py:
```diff
@@ -0,0 +1,46 @@
+        return context
+
+class LogoutView(generic_views.RedirectView):
+
```
Unnecessary empty line.

mitar commented on piplmesh/account/views.py:
```diff
@@ -0,0 +1,46 @@
+from django.views import generic as generic_views
+
+class RegisterView(generic_views.CreateView):
+    template_name = "register.html"
```
Constant string.

mitar commented on piplmesh/frontend/static/piplmesh/css/ie-fix.css:
```diff
@@ -0,0 +1,4 @@
+#nav ul,
```
Unnecessary newline?

byteme replied:
It's easier to see which elements use this property since there will be quite some after we are done with it.

mitar replied:
OK.

mitar commented on piplmesh/frontend/templates/footer.html:
```diff
@@ -1,4 +1,4 @@
-    </div>
+<div id="footer">
+    <div id="license">All content on this website is available under <a href="http://wlan-si.net/en/content-license/">Creative Commons Attribution-ShareAlike license</a> unless specified otherwise.</div>
+    <div id="copyright">&copy; 2012</div>
```
Never hardcode year. Make it programmatic, so that it updates automatically next year. ;-) But in our case this is unnecessary. Such line does not give any additional content to the site (copyright holds automatically, you do not need to write such lines to have copyright).

mitar commented on piplmesh/frontend/templates/header.html:
```diff
@@ -1,16 +1,14 @@
+        <form method="get" action="http://www.google.com/search">
+            <div>
+                <input type="text" name="q" size="25" maxlength="255" value="" />
+                <input type="submit" value="Google Search" />
```
Translatable string.

mitar commented on piplmesh/urls.py:
```diff
@@ -4,10 +4,18 @@
     url(r'^admin/doc/', include('django.contrib.admindocs.urls')),
     url(r'^admin/', include(admin.site.urls)),
+
+    # Authentication
```
Authentication is also a part of frontend, no?

byteme replied:
I wanted to keep it separate (login/logout/register and later on activate and such).

mitar replied:
Then rename (or simply remove the name of) the "frontend" section. So that is confusing then.

byteme replied:
Removed frontend section.

mitar commented:
Not so fast. ;-) Please fix things I commented. And I have commented just types of changes, not everything, so go around and check if there are also similar things to fix elsewhere.

byteme committed:
Code cleaning, trans strings, added jquery and few status icons

mitar commented:
There are still some things you have to fix.

byteme commented:
Yeah, I have already fixed them except the module imports since I didn't know how. I'm gonna update push shortly.

byteme committed:
Fixed module imports

byteme commented:
Ok, I've updated the pull request (with 2 commits).

mitar commented on piplmesh/account/templates/register.html:
```diff
@@ -0,0 +1,17 @@
+
+{% load i18n %}
+
+{% block title %}Register{% endblock %}
```
Translatable string.

byteme replied:
Missed one. I've changed it and updated pull request.

byteme committed:
Added trans string (missed one before).
Remove "frontend" comment in urls.py

byteme commented:
Updated pull request.

mitar commented on piplmesh/account/forms.py:
```diff
@@ -0,0 +1,18 @@
+class RegisterForm(auth_forms.UserCreationForm):
+    class Meta:
+        model = auth_models.User
+        fields = ('username', 'email' )
```
Space.

mitar commented on piplmesh/account/forms.py:
```diff
@@ -0,0 +1,18 @@
+from django.contrib.auth import forms as auth_forms
+from django.contrib.auth import models as auth_models
+from django import forms
```
Please order them alphabetically. And combine into one line:

```
from django import forms
from django.contrib.auth import forms as auth_forms, models as auth_models
```

mitar commented on piplmesh/account/views.py:
```diff
@@ -0,0 +1,33 @@
+from test.threaded_import_hangers import args
```
What is this import?

byteme replied:
I haven't put that there, must have been auto generated by pycharm while i was testing something.

mitar commented on piplmesh/account/views.py:
```diff
@@ -0,0 +1,33 @@
+from django.contrib import auth
+from django import shortcuts
+from django.views import generic as generic_views
+from forms import RegisterForm
```
Import only modules. Use absolute or relative imports. Not import just "forms". You never know what you get (maybe there is some Python package in Python path with this name).

mitar commented on piplmesh/account/views.py:
```diff
@@ -0,0 +1,33 @@
+from test.threaded_import_hangers import args
+from django.contrib import auth
+from django import shortcuts
```
Order alphabetically.

mitar commented on piplmesh/frontend/static/piplmesh/css/style.css:
```diff
@@ -1,64 +1,104 @@
+.container {
+    background: #FFF;
+	margin: auto;
+	width: 960px;
```
Alignment. Be consistent.

byteme replied:
Fixed now

mitar commented on piplmesh/frontend/static/piplmesh/css/style.css:
```diff
@@ -1,64 +1,104 @@
+/* FOOTER STYLE */
+
+#footer {
+	font-size: 10px;
```
Are you using two our four spaces to indent? Be consistent!

mitar commented on piplmesh/frontend/templates/footer.html:
```diff
@@ -1,4 +1,4 @@
-    </div>
+<div id="footer">
+    <div id="license">All content on this website is available under <a href="http://wlan-si.net/en/content-license/">Creative Commons Attribution-ShareAlike license</a> unless specified otherwise.</div>
+    <div id="copyright">&copy; {% now "Y" %}</div>
```
Just remove this line.

byteme replied:
Ok.

mitar replied:
Just the copyright, not license.

mitar commented on piplmesh/urls.py:
```diff
@@ -4,10 +4,20 @@
 from piplmesh.frontend import views as frontend_views
+from piplmesh.account import views as account_views
+
+#test
```
What is this?

byteme replied:
Forgot to remove it. I was testing something.

byteme committed:
Import fixes and coding style changes

byteme committed:
Fixed css alignment problems

byteme committed:
Ordered imports in urls.py

byteme commented:
I've updated pull request, hopefully this is it :)

byteme committed:
Removed copyright div

mitar commented on piplmesh/account/views.py:
```diff
@@ -0,0 +1,32 @@
+from django import shortcuts
+from django.contrib import auth
+from django.views import generic as generic_views
+from piplmesh.account import forms as account_forms
```
Good but group them into sections. Our imports are a separate section (add just one empty line in-between).

byteme replied:
Done.

mitar commented on piplmesh/frontend/templates/header.html:
```diff
@@ -1,16 +1,14 @@
+<div id="header">
+    <div class="logo">
+    <img src="{{ STATIC_URL }}piplmesh/images/logo.png" alt="PiplMesh" />
+    <h1>Pipl Mesh</h1>
```
Write together.

byteme replied:
Done.

mitar commented:
Almost there!

byteme committed:
Separated django imports from custom imports

byteme committed:
Changed Pipl Mesh to PiplMesh

byteme commented:
Ok, I've updated the pull request.

byteme commented:
Updated pull request.

mitar commented:
Can you explain why your team has two pull requests for the same tickets? (#57)

byteme commented:
I've checked that pull request and I don't really see issue 9 (regular registration) there (only lazy signup and facebook).

mitar commented:
Yes, but why are you making two pull requests from the same time? You should coordinate!

mitar commented:
What you mean by regular registration?

byteme commented:
Direct registration, w/o 3rd party (such as facebook).

mitar commented:
For me it seems it has it. ;-)

mitar commented:
OK, now I see what you did. But I thought I was precise enough that we do not want a classical registration. But that we want user to be automatically created and later on just assigned username and password on registration.

But I like your design better than what was before.
